ui/gtk/src/main.rs

```diff
@@ -234,7 +234,7 @@ fn enable_extract_button(archive: Rc<Cell<Option<Archive>>>,
                         for table in arch.tables.iter() {
                             if let Some(data) = table.find_file_contents(name.parse::<u32>().unwrap()) {
                                 let mut output_filepath = dest_dir_path.clone();
-                                output_filepath.push(name.replace("\\", "/"));
+                                output_filepath.push(name.replace("\\", "/") + "." + table.header.file_extension());
```

Comment:
It would probably be better to use [`set_extension`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.set_extension).